relay-server/src/actors/spooler/mod.rs

```diff
@@ -179,148 +176,105 @@ impl FromMessage<RemoveMany> for Buffer {
-                // WAL mode is safe from corruption with synchronous=NORMAL.
-                // When synchronous is NORMAL, the SQLite database engine will still sync at the most critical moments, but less often than in FULL mode.
-                // Which guarantees good balance between safety and speed.
-                .synchronous(SqliteSynchronous::Normal)
-                // The freelist pages are moved to the end of the database file and the database file is truncated to remove the freelist pages at every
-                // transaction commit. Note, however, that auto-vacuum only truncates the freelist pages from the file.
-                // Auto-vacuum does not defragment the database nor repack individual database pages the way that the VACUUM command does.
-                //
-                // This will helps us to keep the file size under some control.
-                .auto_vacuum(SqliteAutoVacuum::Full)
-                // If shared-cache mode is enabled and a thread establishes multiple
-                // connections to the same database, the connections share a single data and schema cache.
-                // This can significantly reduce the quantity of memory and IO required by the system.
-                .shared_cache(true);
-
-            let db = SqlitePoolOptions::new()
-                .max_connections(config.spool_envelopes_max_connections())
-                .min_connections(config.spool_envelopes_min_connections())
-                .connect_with(options)
-                .await
-                .map_err(BufferError::SetupFailed)?;
+    /// Returns the number of envelopes in the memory buffer.
+    fn count(&self) -> usize {
+        self.buffer.values().map(|v| v.len()).sum()
+    }
 
-            let spool_config = BufferSpoolConfig {
-                db,
-                max_disk_size: config.spool_envelopes_max_disk_size(),
-                max_memory_size: config.spool_envelopes_max_memory_size(),
-                is_disk_full: false,
-            };
+    /// Removes envelopes from the in-memory buffer.
+    fn remove(&mut self, keys: &BTreeSet<QueueKey>) -> usize {
+        let mut count = 0;
+        for key in keys {
+            let (current_count, current_size) = self.buffer.remove(key).map_or((0, 0), |k| {
+                (k.len(), k.into_iter().map(|k| k.estimated_size()).sum())
+            });
+            count += current_count;
+            self.used_memory -= current_size;
+        }
+        relay_statsd::metric!(
+            histogram(RelayHistograms::BufferEnvelopesMemory) = self.used_memory as f64
+        );
+
+        count
+    }
 
-            service.spool_config = Some(spool_config);
+    /// Dequeues the envelopes from the in-memory buffer and send them to provided `sender`.
+    fn dequeue(&mut self, keys: &Vec<QueueKey>, sender: mpsc::UnboundedSender<ManagedEnvelope>) {
+        for key in keys {
+            for value in self.buffer.remove(key).unwrap_or_default() {
```

Comment:
This makes it easier to understand it:
```suggestion
            for envelope in self.buffer.remove(key).unwrap_or_default() {
```